RevenueCat/Scripts/Purchases.cs

```diff
@@ -58,7 +58,26 @@ public partial class Purchases : MonoBehaviour
 
     [Tooltip("A subclass of Purchases.UpdatedCustomerInfoListener component.\n" +
              "Use your custom subclass to define how to handle updated customer information.")]
-    public UpdatedCustomerInfoListener listener;
+    [SerializeField]
+    private UpdatedCustomerInfoListener listener;
+    
+    // Cache custom listener, if not from class.
+    private IUpdatedCustomerInfoListener customListener;
+    public IUpdatedCustomerInfoListener Listener
```

Comment:
I believe this means that devs setting the listener through code before might experience a breaking change.
- Before:
```
purchases.listener = CustomListener()
```

- After:
```
purchases.Listener = CustomListener()
```

I do think that most people would set the listener through the editor, considering it had to be a `MonoBehavior`... But I think we might need to make the `listener` public private still to avoid the breaking change, and encourage people to move to use `Listener` instead. Wdyt?

Reply:
Maybe we can hold on this change until the next major as well, and release the change below

Reply:
You're correct, this is breaking change in code. I decided for this approach for two reasons:
- It's very easy to search and replace in code: `Listener` instead of `listener`. Could leave some comments and hint maybe for those people that use it in code.
- It's very clear from API design perspective that the private field is meant for Editor only and you only allow setting it there. In other way you would have two exposed `listener` and `Listener` in the API, which could get confusing, if in fact you did want to set it in code as a new user.

My suggestion would be to keep the API clearer and rather make a major release (or however you think is best), if this breaking change is a problem, since this move will make the API clearer in the future. But at the end it's your call how you want to have it, definitely both solutions work for our case. Maybe a better approach would be to have a `List` of those listeners instead and keep the original one public and just add like a `AddListener`/`RemoveListener` methods to manage the list.. Just thinking out loud.

Your call, let me know if you need me to change in the PR.

Reply:
Yeah I agree that it would be better to ship this in a major, rather than keeping both which would cause a very confusing API.

If you want to revert this change and leave the one below, we would be happy to get that in!

As for your proposal of having `AddListener/`RemoveListener`, I think that could work! We actually allow doing this in some other SDKs. But seems that could be worked on separately. Thanks for the suggestion!

Reply:
I can also do the `Add/Remove`Listener if you like, but not sure which change you want me to revert? The change to `private`?

Reply:
Sorry for not being clear! My suggestion is to:
- Revert the changes related to the listener (including the new interface)
- Keep the changes related to initialization of the SDK at different points of time.

Then merge the PR with just that second change, which shouldn't be a breaking change. And we can release that soon.

As for doing the `Add/Remove` listener, I think that works too! And it would be amazing if you can do it 👼. The only point is to avoid breaking changes (at least until we're ready to make another major)

Please let me know if you have questions or help with the listener approach! We also think that would be a good addition 👍

Reply:
Cool, I understand, I'll get on it this week.